model/work.py

```diff
@@ -1441,10 +1453,11 @@ def query_to_json_array(query):
         ).where(
             and_(
                 LicensePool.work_id==work_id_column,
+                work_presentation_edition_id_column==Edition.id,
```

Comment:
The absence of this little bit causes a bunch of problems that aren't caught by test_search_document, because that method only tests one work. So I added the "just a basic check" test in test_external_search to check for overzealous joins in general.